File: app/src/main/java/com/dpis/module/SettingsUiState.java
package com.dpis.module;

/** Immutable presentation snapshot; persistence and workflow execution stay Java-owned. */
public final class SettingsUiState {
    public final boolean storeAvailable;
    public final boolean systemHooksEnabled;
    public final boolean safeModeEnabled;
    public final boolean globalLogEnabled;
    public final boolean launcherIconHidden;
    public final int interfaceScalePercent;
    public final boolean cacheClearInProgress;
    public final String cacheUsage;
    public final String languageLabel;

    SettingsUiState(boolean storeAvailable, boolean systemHooksEnabled, boolean safeModeEnabled,
            boolean globalLogEnabled, boolean launcherIconHidden, int interfaceScalePercent,
            boolean cacheClearInProgress, String cacheUsage, String languageLabel) {
        this.storeAvailable = storeAvailable;
        this.systemHooksEnabled = systemHooksEnabled;
        this.safeModeEnabled = safeModeEnabled;
        this.globalLogEnabled = globalLogEnabled;
        this.launcherIconHidden = launcherIconHidden;
        this.interfaceScalePercent = interfaceScalePercent;
        this.cacheClearInProgress = cacheClearInProgress;
        this.cacheUsage = cacheUsage != null ? cacheUsage : "";
        this.languageLabel = languageLabel != null ? languageLabel : "";
    }
}

File: app/src/main/java/com/dpis/module/settings/SettingsUiState.kt
package com.dpis.module.settings

import android.net.Uri

/** Immutable presentation snapshot; persistence and workflow execution stay Java-owned. */
class SettingsUiState(
    @JvmField val storeAvailable: Boolean,
    @JvmField val systemHooksEnabled: Boolean,
    @JvmField val safeModeEnabled: Boolean,
    @JvmField val globalLogEnabled: Boolean,
    @JvmField val launcherIconHidden: Boolean,
    @JvmField val interfaceScalePercent: Int,
    @JvmField val cacheClearInProgress: Boolean,
    cacheUsage: String?,
    languageLabel: String?,
    @JvmField val pendingImportUri: Uri?,
) {
    @JvmField val cacheUsage: String = cacheUsage ?: ""
    @JvmField val languageLabel: String = languageLabel ?: ""
}
